src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -296,103 +313,407 @@ module GenericSimple = {
   };
 };
 
-module PointwiseAddDistributionsWeighted = {
-  type t = pointwiseAdd;
+module DistTree = {
+  type nodeResult = [
```

Comment:
why is this called ``nodeResult``? Can it just be ``node``? It seems like ``nodeResult``s are the inputs to some functions.

Reply:
I'm also confused at what this is doing; why do we need to return a distribution in some cases, as opposed to the rendered version in all cases?

Reply:
Right, the idea was to delay rendering as much as possible, so that e.g. `normal + normal` can directly evaluate to another `normal`, without having to do convolution on xyShapes.

Reply:
Ah, cool. Maybe nodeResult could be called something like, ``expressiveNode``, which is different from ``renderedNode``; ``expressiveNode`` is the same as ``renderedNode`` (both have errors), but ``expressiveNode`` contains these extra types for this purpose. Then this could be documented above. Another phrase would be ``renderedPlusSimpleNode`` or something.

Reply:
Can you show me an example of what you have in mind? Also I would like the name to somehow express that these node things contain the result of a subtree having been evaluated, as opposed to them being nodes in the original distTree. Maybe `node` isn't even a good term here?

Reply:
I'm not big fan of the word "evaluated" because it is an incredibly generic term. Anytime there's a transformation between one type and another, you could argue that there was an evaluation that happened in between.

It does seem awkward to use the word ``node`` in some areas and ``tree`` in others; we should choose one throughout the naming.

If this ``nodeResult`` type only exists in a certain module, and will not be referred by other modules, it's less of an issue.  Maybe even something like, we have ``node``(currently called distTree) and we have ``partiallyRenderedNode``.

Reply:
I see what you mean. Perhaps it isn't even necessary to have two different types, as long as there is a `` `Render`` operation at the very top to ensure that only a `` `RenderedShape`` kind of `node` comes out at the end. I'm sure I can come up with something cleaner here.

Reply:
I like the idea of adding a `Render at top. That way other functions could choose not to do this step if they want, hypothetically.